src/middleware.ts

```diff
@@ -68,7 +68,14 @@ export async function middleware(req: NextRequest, ev: NextFetchEvent) {
 		// Next.js doesn't support redirecting to a pathname, so we clone the
 		// request URL to adjust the pathname in an absolute URL
 		const url = req.nextUrl.clone()
-		url.pathname = destination
+
+		// Simple redirects _can_ contain a hash, which we need to detect and handle
+		const [pathname, hash] = destination.split('#')
+		url.pathname = pathname
+		if (hash) {
+			url.hash = hash
+		}
```

Comment:
This was an uncaught edge case with how we handle our "simple redirects" that are delegated to middleware. Redirects with query params are not handle here, so it seems reasonable to only worry about the hash.